keps/sig-node/4680-add-resource-health-to-pod-status/README.md

```diff
@@ -245,7 +295,39 @@ The API will be limited to "prepared" devices and include the claim `name/namesp
 
 Kubelet will react on this field the same way as we propose to do it for the Device Plugin.
 
-Specific implementation details will be added for the beta.
+The new method will be added to [Node service](https://github.com/kubernetes/kubernetes/blob/04bba3c222bb2c5b1b1565713de4bf334ee7fbe4/staging/src/k8s.io/kubelet/pkg/apis/dra/v1alpha4/api.proto#L34)
+alongside `NodePrepareResources` and `NodeUnprepareResources` and a `health` field for `Device` structure:
+
+``` proto
+service Node {
+  ...
+
+	// WatchDevicesStatus returns a stream of List of Devices
+	// Whenever a Device state change or a Device disappears, WatchDevicesStatus
+	// returns the new list.
+  // This method is optional and may not be implemented.
+	rpc WatchDevicesStatus(Empty) returns (stream DevicesStatusResponse) {}
+}
```

Comment:
Can you put this into a new gRPC interface? Then it can be versioned as part of this KEP and drivers are not forced to implement this method - they would be if it became part of the [NodeServer interface](https://github.com/kubernetes/kubernetes/blob/d88b4e3b6e34a85f58778b7ef96e64edffff6823/staging/src/k8s.io/kubelet/pkg/apis/dra/v1alpha4/api.pb.go#L590-L600).

At a practical level it'll help avoid code conflicts. I need to promote that dra/v1alpha4 API to dra/v1beta1 in 1.32.

Reply:
Is there any implications of having two interfaces running? Would it mean we need an extra port for the new interface?

Reply:
You can serve different gRPC interfaces over the same connection. They are like different endpoints in an HTTP server.

That we version them is a bit odd. There's no difference between v1alpha4 and v1beta1, but because we put the version into the name, they are completely different as far as gRPC is concerned. But that's just an aside.

Reply:
I'm okay with treating this as an implementation detail that'll get addressed later - no need to update the KEP for it now.

Reply:
added a note about it.